include/openenclave/internal/raise.h

```diff
@@ -134,18 +134,10 @@ OE_EXTERNC_BEGIN
         result = (RESULT);                                           \
         if (result != OE_OK)                                         \
         {                                                            \
-            if (!_strcmp(#__VA_ARGS__, "NULL"))                      \
```

Comment:
Newer MSVC releases support a conformant preprocessor: https://devblogs.microsoft.com/cppblog/announcing-full-support-for-a-c-c-conformant-preprocessor-in-msvc/
https://devblogs.microsoft.com/cppblog/msvc-preprocessor-progress-towards-conformance/

which handles __VA_ARGS__ correctly, and _MSVC_TRADITIONAL could be used to decide when to enable the workaround.

Reply:
Thanks for the information! The question then is when we should update to use newer MSVC. In the meantime, we still need to maintain the backward compatibility.

Reply:
Yes, absolutely, and keeping the current code is probably simpler than ifdef’ing on _MSVC_TRADITIONAL. Perhaps adding a comment would be a good thing though?

Reply:
@mingweishih I believe that was the intent of the implementation, but I don't believe it works in practice and was never tested. All of the existing OE_RAISE_MSG instances in host/common code still have the "NULL" vararg added explicitly, and if you try to remove one of them, the MSVC build fails. (i.e. gets processed to `_strcmp(, "NULL")` which is illegal).

Reply:
@CodeMonkeyLeet Yes, what I'm trying to point out is that the IF check here is necessary for keeping the compatibility between MSVC and gcc/clang so should not be removed.

Reply:
@mingweishih Yes, and I'm pointing out that _it does not do that correctly_ so keeping it around is redundant. The PR that introduced this change was never tested. If you try to use OE_RAISE_MSG without varargs on Windows, it still breaks the build (with the existing version of MSVC in our CI)

I'm still of the opinion that this macro is unnecessarily complicated, and should be more generally simplified.

Reply:
Isn't the removed check in this PR only for the non-MSVC case (i.e., trying to remove last `NULL` argument if present with gcc/clang)?

Reply:
@mingweishih I apologize, you're right. I'm confusing the line being commented here with line 113: `(!_strcmp(#__VA_ARGS__, ""))` which is what I was referring to as not working in MSVC.

My comment about this macro being unnecessarily complicated still applies though: that we need to have two different versions of it for different compilers and attempting to accomodated both usages with and without a `NULL` varargs seems like it should be simplified down, not just addressing the specific static analysis issue here (which only seems to apply to the linux build?)

Reply:
To I think support Simon's point, since 137 is in the NON-msvc case, why is a strcmp needed at all in this line?

Reply:
It's because we're using the pattern of  `OE_RAISE_MSG(..., ..., NULL)` on the host-side, which should be compatible with both MSRC and clang/gcc. See https://github.com/openenclave/openenclave/blob/master/host/sgx/create.c#L810 for an example.

Reply:
Please add a comment in the file about why this is needed (and that comment should not mention MSVC since it's irrelevant in this path).

Reply:
This has been explained in line 89 - 103 already.

Reply:
Thanks! I will leave the macro as is and figure out a way to suppress the codeql alert or mark false positive.
Closing this PR.